### PR TITLE
[chore] fix process scraper test to not run on Darwin and Windows for cgroups

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Fix tests on Mac for process scraper.

### CODE REVIEW

File: receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go [hunk 1]
```diff
@@ -722,15 +722,15 @@ func TestScrapeMetrics_ProcessErrors(t *testing.T) {
 		},
 		{
 			name:        "Cgroup Error",
-			osFilter:    "linux",
+			osFilter:    []string{"darwin", "windows"},
```
[the remark this answers is not in the document]
Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:725]: Yes. This is the fix. The osFilter filters _out_ which OS to run in. So it was excluding linux from running.

File: receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go [hunk 2]
```diff
@@ -193,7 +193,7 @@ func TestScrape(t *testing.T) {
 				expectedMetricCount += 4
 			}
 			if test.expectConnectionsMetric {
-				expectedMetricCount++
+				expectedMetricCount += 2
```
Comment (astencel-sumo) [inline receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:196]: Why does this change? Doesn't it break the tests?

Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:196]: let's revert and find out

Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go:196]:
```suggestion
				expectedMetricCount++
```

Comment (atoulme) [reply receiver/hostmetricsreceiver/internal/scraper/networkscraper/network_scraper_test.go]: that was totally it. Thank you.

**Review by andrzej-stencel — APPROVED** [review]

**Review by djaglowski — APPROVED** [review]

**Pull request merged**